cirq/protocols/resolve_parameters.py

```diff
@@ -154,7 +156,17 @@ def resolve_parameters(
         return type(val)(resolve_parameters(e, param_resolver, recursive) for e in val)
 
     getter = getattr(val, '_resolve_parameters_', None)
-    result = NotImplemented if getter is None else getter(param_resolver, recursive)
+    try:
+        result = NotImplemented if getter is None else getter(param_resolver, recursive)
+    except TypeError:
```

Comment:
I would be very explicit about the exact error message here. Otherwise other type of implementation errors resulting in `TypeError` will be handled by this logic, which will be confusing as the real `TypeError` will be then swallowed. 

In particular I would narrow the handling of this branch in case we get this specific error string: `_resolve_parameters_() takes 2 positional arguments but 3 were given`

Reply:
Opted to avoid `try/except` altogether - see Doug's comment.